lib/Frontend/Frontend.cpp

```diff
@@ -557,24 +698,26 @@ void CompilerInstance::performSema() {
                             options.WarnLongFunctionBodies,
                             options.WarnLongExpressionTypeChecking,
                             options.SolverExpressionTimeThreshold);
+}
 
-  // Even if there were no source files, we should still record known
-  // protocols.
-  if (auto *stdlib = Context->getStdlibModule())
-    Context->recordKnownProtocols(stdlib);
-
-  if (DelayedCB) {
-    performDelayedParsing(MainModule, PersistentState,
-                          Invocation.getCodeCompletionFactory());
-  }
-
-  // Perform whole-module type checking.
+void CompilerInstance::typeCheckMainModule(
```

Comment:
It seems odd to split this out. Type-checking the files in the main module is part of type-checking the main module. See [docs/Lexicon.rst](https://github.com/apple/swift/blob/master/docs/Lexicon.rst).